source-braintree-native/source_braintree_native/api.py

```diff
@@ -0,0 +1,253 @@
+from braintree import (
+    BraintreeGateway,
+    AddOnGateway,
+    Configuration,
+    TransactionSearch,
+    CustomerSearch,
+    CreditCardVerificationSearch,
+    DisputeSearch,
+    SubscriptionSearch,
+    )
+from braintree.attribute_getter import AttributeGetter
+from datetime import datetime, timedelta, UTC
+from logging import Logger
+from typing import AsyncGenerator
+from estuary_cdk.capture.common import LogCursor
+
+from .models import (
+    FullRefreshResource,
+    Customer,
+    Dispute,
+    Subscription,
+    Transaction,
+    CreditCardVerification,
+)
+
+
+# Searches return at most 10,000 results (50,000 for transaction searches). If we hit this limit,
+# the connector could have missed data and we'll need to use smaller date windows.
+SEARCH_LIMIT = 10_000
+TRANSACTION_SEARCH_LIMIT = 50_000
+# Incremental streams use sliding date windows to try & avoid hitting the above search limits.
+WINDOW_SIZE = 15
+
+CONVENIENCE_OBJECTS = [
+    'gateway'
+]
+
+
+def _braintree_object_to_dict(braintree_object):
```

Comment:
Just a sanity check: top-level entities in Braintree are always objects?

Reply:
Yep, their SDK returns an object, like the [Transaction](https://developer.paypal.com/braintree/docs/reference/response/transaction/python) object.